model/transactions/payment.go

```diff
@@ -8,13 +8,73 @@ import (
 
 type Payment struct {
 	BaseTx
-	Amount         types.CurrencyAmount
-	Destination    types.Address
-	DestinationTag uint                 `json:",omitempty"`
-	InvoiceID      uint                 `json:",omitempty"`
-	Paths          [][]PathStep         `json:",omitempty"`
-	SendMax        types.CurrencyAmount `json:",omitempty"`
-	DeliverMin     types.CurrencyAmount `json:",omitempty"`
+	/*
+		API v1: Only available in API v1.
+		The maximum amount of currency to deliver.
+		For non-XRP amounts, the nested field names MUST be lower-case.
+		If the tfPartialPayment flag is set, deliver up to this amount instead.
+	*/
+	Amount types.CurrencyAmount
+	/*
+		API v2: Only available in API v2.
+		The maximum amount of currency to deliver.
+		For non-XRP amounts, the nested field names MUST be lower-case.
+		If the tfPartialPayment flag is set, deliver up to this amount instead.
+	*/
+	DeliverMax types.CurrencyAmount `json:",omitempty"`
+	/*
+		(Optional) Minimum amount of destination currency this transaction should deliver.
+		Only valid if this is a partial payment.
+		For non-XRP amounts, the nested field names are lower-case.
+	*/
+	DeliverMin types.CurrencyAmount `json:",omitempty"`
+	/*
+		The unique address of the account receiving the payment.
+	*/
+	Destination types.Address
+	/*
+		(Optional) Arbitrary tag that identifies the reason for the payment to the destination, or a hosted recipient to pay.
+	*/
+	DestinationTag uint `json:",omitempty"`
+	/*
+		(Optional) Arbitrary 256-bit hash representing a specific reason or identifier for this payment
+	*/
+	InvoiceID uint `json:",omitempty"`
+	/*
+		(Optional, auto-fillable) Array of payment paths to be used for this transaction.
+		Must be omitted for XRP-to-XRP transactions.
+	*/
+	Paths [][]PathStep `json:",omitempty"`
+	/*
+		(Optional) Highest amount of source currency this transaction is allowed to cost,
+		including transfer fees, exchange rates, and slippage.
+		Does not include the XRP destroyed as a cost for submitting the transaction.
+		For non-XRP amounts, the nested field names MUST be lower-case.
+		Must be supplied for cross-currency/cross-issue payments.
+		Must be omitted for XRP-to-XRP payments.
+	*/
+	SendMax types.CurrencyAmount `json:",omitempty"`
+}
+
+// "New" creates a new Payment object based on the provided Payment struct.
+// It validates the different fields of the payment.
+// If no validation error is found, it initializes a new Payment object with the provided fields and returns it.
+func New(payment Payment) Payment {
```

Comment:
After reviewing the repo, I have been thinking about whether or not to add a `New` function for each type of transaction. 

I have found two solutions that I find interesting:

1. Add for each transaction a `NewXXX` function that initializes the `TransactionType` of the `BaseTx` to the corresponding one and add set methods for each field (which adds a lot of verbosity and it is not worth it because the fields are accessible).

2. Do not use a `NewXXX` function for the fields. Keep the current struct with the accessible fields and do not validate the transaction.

In my opinion, the second approach fits better since it does not add more verbosity and allows to create partial transactions so that they can be updated for the user when needed.

What seems right to me is to add a validation function for each transaction to be used later for unmarshalling or signing transactions.

Wdyt?

Reply:
The second option sounds good to me. I agree that a `New` function for every Struct adds a lot of verbosity. I will remove it for the `Payment` transaction and make the Struct fields accessible (uppercase if I am correct).